gui/default/syncthing/core/syncthingController.js

```diff
@@ -1856,37 +1935,46 @@ angular.module('syncthing.core')
 
         $scope.addFolder = function () {
             $http.get(urlbase + '/svc/random/string?length=10').success(function (data) {
-                $scope.editingExisting = false;
-                $scope.currentFolder = angular.copy($scope.folderDefaults);
-                initShareEditing('folder');
-                $scope.currentFolder.id = (data.random.substr(0, 5) + '-' + data.random.substr(5, 5)).toLowerCase();
-                $scope.currentSharing.unrelated = $scope.otherDevices();
-                $scope.ignores.text = '';
-                $scope.ignores.error = null;
-                $scope.ignores.disabled = false;
-                $scope.editFolderModal();
+                var folderID = (data.random.substr(0, 5) + '-' + data.random.substr(5, 5)).toLowerCase();
+                addFolderInit(folderID).then(function() {
+                    // Triggers the watch that sets the path
+                    $scope.currentFolder.label = $scope.currentFolder.label;
+                    editFolderModal();
+                });
             });
         };
 
-        $scope.addFolderAndShare = function (folder, folderLabel, device) {
-            $scope.editingExisting = false;
-            $scope.currentFolder = angular.copy($scope.folderDefaults);
-            $scope.currentFolder.id = folder;
-            $scope.currentFolder.label = folderLabel;
-            $scope.currentFolder.viewFlags = {
-                importFromOtherDevice: true
-            };
-            initShareEditing('folder');
-            $scope.currentSharing.selected[device] = true;
-            $scope.currentSharing.unrelated = $scope.deviceList().filter(function (n) {
-                return n.deviceID !== $scope.myID && !$scope.currentSharing.selected[n.deviceID]
+        $scope.addFolderAndShare = function (folderID, folderLabel, device) {
+            addFolderInit(folderID).then(function() {
+                $scope.currentFolder.viewFlags = {
+                    importFromOtherDevice: true
+                };
+                $scope.currentSharing.selected[device] = true;
+                $scope.currentFolder.label = folderLabel;
+                editFolderModal();
             });
-            $scope.ignores.text = '';
-            $scope.ignores.error = null;
-            $scope.ignores.disabled = false;
-            $scope.editFolderModal();
         };
 
+        function addFolderInit(folderID) {
+            $scope.editingExisting = false;
+            $scope.editingDefaults = false;
+            return $http.get(urlbase + '/config/defaults/folder').then(function(p) {
+                $scope.currentFolder = p.data;
+                for (var k in $scope.versioningDefaults) {
```

Comment:
Why are these not stored in the folder config?

Reply:
Two parts: One was a bug in the current PR, the other is a conscious omission:  
Bug: The default versioning type including params is stored in `config.defaults.folder.versioning`, however wasn't applied when adding a new folder. That's now fixed, along with moving the gui-only parameters into `currentFolder._guiVersioning`.  
Omission: Versioning parameters are currently "free-form" (`map[string]string`), which prevents using `ext.default` in proto. The consistent thing to do would be creating proto types for all of these and adding a `Config.Defaults.VersioningParams` with a new map-like struct (type -> params). That just feels like a lot of work for little gain (defaults for something that's off by default).